pkg/domain/vulnerability.go

```diff
@@ -156,3 +156,84 @@ type WebVulnerability struct {
 	CreatedAt       *time.Time `json:"created_at"`
 	UpdatedAt       *time.Time `json:"updated_at"`
 }
+
+type CWEDetail struct {
+	ID          string     `json:"cwe_id"`
+	Title       string     `json:"title"`
+	Description string     `json:"description"`
+	CreatedAt   *time.Time `json:"created_at"`
+	LastUpdated *time.Time `json:"last_updated"`
+}
+
+// GeneratePortDataFromWebVuln creates PortData based on WebVulnerability information
+func GeneratePortDataFromWebVuln(vuln tools.WebVulnerability) tools.PortData {
+	portData := tools.PortData{
+		ID:       80,
+		Protocol: "tcp",
+		Service: tools.Service{
+			Name: "http",
+		},
+		State:           "open",
+		Vulnerabilities: nil,
+	}
+	if vuln.WascID == "45" {
+		portData.ID = 443
+		portData.Protocol = "tcp"
+		portData.Service.Name = "nginx"
+	}
+	return portData
+}
+
+// GenerateVulnFromWebVuln creates a tools.Vulnerability from WebVulnerability and Scan data
+func GenerateVulnFromWebVuln(scan *Scan, vuln tools.WebVulnerability) tools.Vulnerability {
+	today := time.Now()
+	return tools.Vulnerability{
+		HostID:             scan.HostID,
+		ScanID:             scan.ID,
+		CweID:              vuln.CweID,
+		Type:               enums.OwaspCategoryNoInfo,
```

Comment:
Si vamos a poblar este campo, deberia ser:

```suggestion
		Type:               enums.GetOwaspCategoryForCWE(vuln.CweID),
```

Reply:
Thanks for the correction